data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/ParserTest.java

```diff
@@ -0,0 +1,102 @@
+    private static final Class<? extends ParseTree> RelationalOperatorExpression = DataPrepperExpressionParser.RelationalOperatorExpressionContext.class;
+    private static final Class<? extends ParseTree> RelationalOperator = DataPrepperExpressionParser.RelationalOperatorContext.class;
+    private static final Class<? extends ParseTree> SetOperatorExpression = DataPrepperExpressionParser.SetOperatorExpressionContext.class;
+    private static final Class<? extends ParseTree> SetOperator = DataPrepperExpressionParser.SetOperatorContext.class;
+    private static final Class<? extends ParseTree> UnaryOperatorExpression = DataPrepperExpressionParser.UnaryOperatorExpressionContext.class;
+    private static final Class<? extends ParseTree> ParenthesisExpression = DataPrepperExpressionParser.ParenthesisExpressionContext.class;
+    private static final Class<? extends ParseTree> RegexPattern = DataPrepperExpressionParser.RegexPatternContext.class;
+    private static final Class<? extends ParseTree> SetInitializer = DataPrepperExpressionParser.SetInitializerContext.class;
+    private static final Class<? extends ParseTree> UnaryNotOperatorExpression = DataPrepperExpressionParser.UnaryNotOperatorExpressionContext.class;
+    private static final Class<? extends ParseTree> UnaryOperator = DataPrepperExpressionParser.UnaryOperatorContext.class;
+    private static final Class<? extends ParseTree> Primary = DataPrepperExpressionParser.PrimaryContext.class;
+    private static final Class<? extends ParseTree> JsonPointer = DataPrepperExpressionParser.JsonPointerContext.class;
+    private static final Class<? extends ParseTree> VariableIdentifier = DataPrepperExpressionParser.VariableIdentifierContext.class;
+    private static final Class<? extends ParseTree> VariableName = DataPrepperExpressionParser.VariableNameContext.class;
+    private static final Class<? extends ParseTree> Literal = DataPrepperExpressionParser.LiteralContext.class;
+
+    @Mock
+    private TokenStream tokenStream;
+    @InjectMocks
+    private DataPrepperExpressionParser parser;
+
+    private void withTokenStream(final Integer ... types) {
+        final MockTokenStreamHelper helper = new MockTokenStreamHelper(types);
+
+        doAnswer(helper::consume).when(tokenStream).consume();
+        doAnswer(helper::LT).when(tokenStream).LT(anyInt());
+        doAnswer(helper::LA).when(tokenStream).LA(anyInt());
+
+        parser.setInputStream(tokenStream);
+    }
+
+    @Test
+    void testEqualityExpression() {
+        withTokenStream(
+                DataPrepperExpressionParser.Integer,
+                DataPrepperExpressionParser.EQUAL,
+                DataPrepperExpressionParser.Float,
+                DataPrepperExpressionParser.EOF
+        );
+
+        final DataPrepperExpressionParser.ExpressionContext expression = parser.expression();
+
+        assertThat(expression, isParseTree(Expression).withChildrenMatching(
```

Comment:
I'm a little concerned that these tests can become unreadable as they grow in complexity.

Is there a way to break these out into multiple individual `assertThat` statements?

Reply:
I like the idea of breaking down the larger assertion and open to ideas on how we could breakup the large assertion. An alternative is to split the assertions per node. It would look something like:

```
assertThat(expression, is(instanceOf(DataPrepperExpressionParser.ExpressionContext.class)));
assertThat(expression.getChildCount(), is(2));

final ParseTree conditionalExpression = expression.getChild(0);
assertThat(conditionalExpression, is(instanceOf(DataPrepperExpressionParser.ConditionalExpressionContext.class)));
assertThat(conditionalExpression.getChildCount(), is(1));

final ParseTree eofTerminalNode = expression.getChild(1);
assertThat(eofTerminalNode, is(instanceOf(TerminalNode.class)));

final ParseTree equalityExpression1 = conditionalExpression.getChild(0);
assertThat(equalityExpression1, is(instanceOf(DataPrepperExpressionParser.EqualityOperatorExpressionContext.class)));
assertThat(equalityExpression1.getChildCount(), is(3));

final ParseTree equalityExpression2 = conditionalExpression.getChild(0);
assertThat(equalityExpression2, is(instanceOf(DataPrepperExpressionParser.EqualityOperatorExpressionContext.class)));
assertThat(equalityExpression2.getChildCount(), is(1));

final ParseTree equalityOperator = conditionalExpression.getChild(1);
assertThat(equalityOperator, is(instanceOf(DataPrepperExpressionParser.EqualityOperatorContext.class)));
assertThat(equalityOperator.getChildCount(), is(1));

final ParseTree regexOperatorExpression1 = conditionalExpression.getChild(0);
assertThat(regexOperatorExpression1, is(instanceOf(DataPrepperExpressionParser.RegexOperatorExpressionContext.class)));
assertThat(regexOperatorExpression1.getChildCount(), is(1));

final ParseTree regexOperatorExpression2 = equalityExpression2.getChild(0);
assertThat(regexOperatorExpression2, is(instanceOf(DataPrepperExpressionParser.RegexOperatorExpressionContext.class)));
assertThat(regexOperatorExpression2.getChildCount(), is(1));

// ...
```
I expect that to quickly become unreadable.